### DISCUSSION TITLE
Pulumi state files in VCS - what to include?

### DISCUSSION THREAD

**Discussion by joaodlf** [pulumi/pulumi#16269] [General]
Simple question really - if I wanted to manage my state via VCS, which files do I want to include and which ones to ignore via .gitignore?

Usage: centralised repo that hosts the pulumi code + state, managed via Pull Requests that automatically run `pulumi up` when merged. (I am not looking for a discussion on best practices, cloud vs local, etc. This a very simple setup just for myself).

The .pulumi dir includes quite a lot of files, some clarity would be helpful. Is it just the files within `stacks` that are relevant? What about the `history` dir?

**Comment by fingon** [comment]
You can get one file dumps of stacks using export. Perhaps importing one, running up, and then exporting it could work? ( I am thinking of doing something like that, with pulumi preview on PR to see the change before merge would apply it ).
